torch/_subclasses/meta_utils.py

```diff
@@ -1206,90 +1305,65 @@ def __call__(
-                                symbolic_context=symbolic_context,
-                            )
-                        out = torch._to_functional_tensor(fake_t)
-                        torch._mirror_autograd_meta_to(fake_t, out)
-                        return out
-                    else:
-                        # torch.func.functionalize
-                        reapply_views = torch._C._functionalization_reapply_views_tls()
-                        unwrap_t = _unwrap_functional_tensor(t, reapply_views)
-                        pop_st_ctx = (
-                            torch._functorch.pyfunctorch.temporarily_pop_interpreter_stack()
-                        )
-                        with pop_st_ctx:
-                            fake_t = self.meta_tensor(
-                                self.describer.describe_tensor(unwrap_t),
-                                shape_env=shape_env,
-                                callback=callback,
-                                source=source,
-                                symbolic_context=symbolic_context,
-                            )
-                        return _wrap_functional_tensor(fake_t, current_level())
                 self.miss += 1
                 return NotImplemented
             else:
                 self.hit += 1
-
-                disable_functorch = torch._C._DisableFuncTorch
-                with disable_functorch():
-                    r = self.meta_tensor(
-                        self.describer.describe_tensor(t),
-                        shape_env=shape_env,
-                        callback=callback,
-                        source=source,
-                        symbolic_context=symbolic_context,
-                    )
-                if type(t) is torch.nn.Parameter:
-                    # NB: Cannot directly use Parameter constructor
-                    # because that would force a detach, not desirable
-                    r._is_param = True
-                return r
         elif torch.overrides.is_tensor_like(t):
             self.miss += 1
             return NotImplemented
         else:
             # non-Tensor types don't count as hit or miss
             return t
 
+        # Describe the tensor.  NB: do NOT disable ambient modes, we may need
+        # to query them when figuring out what to put in here
+        t_desc = self.describer.describe_tensor(t)
+
+        # Do the meta-fication.  Here, we disable all the ambient modes, to
+        # better simulate what would be like to re-fakeify from a fresh
+        # process
+        with contextlib.ExitStack() as exit_stack:
+            exit_stack.enter_context(torch._dispatch.python.suspend_functionalization())
+            st = peek_interpreter_stack()
+            if st is not None:
+                exit_stack.enter_context(
+                    torch._functorch.pyfunctorch.temporarily_clear_interpreter_stack()
+                )
+            exit_stack.enter_context(torch._C._DisableFuncTorch())
```

Comment:
We shouldn't need this anymore -- there should already exist an invariant that if the interpreter stack is empty, the functorch keys are disabled. Is that not the case?

Reply:
I'm just preserving the old _DisableFuncTorch. I can try removing it in a follow up PR

Reply:
sg